### ISSUE TITLE
mounts hang when efs-csi-node pods are restarted because of empty privateKey.pem

### ISSUE THREAD

**Issue by universam1** [kubernetes-sigs/aws-efs-csi-driver#683]
/kind bug

**What happened?**

The same issue as #178 and #569, still not solved.

After the EKS driver container is replaced (i.e. by terminating the driver process or upgrading the driver to a new image), all existing mounts on that node hang for 1 hour

```
Warning  FailedMount  22m (x62 over 4h59m)   kubelet  Unable to attach or mount volumes: unmounted volumes=efs-data

Warning  FailedMount  6m54s (x81 over 5h2m)  kubelet  MountVolume.SetUp failed for volume "xxx-efs" : kubernetes.io/csi: mounter.SetUpAt failed to check for STAGE_UNSTAGE_VOLUME capability: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial unix /var/lib/kubelet/plugins/efs.csi.aws.com/csi.sock: connect: connection refused
mount failed: exit status 1

Mounting command: mount
Mounting arguments: -t efs -o accesspoint=fsap-xxxx,tls,noatime fs-xxxxx:/ /var/lib/kubelet/pods/xxxxx/volumes/kubernetes.io~csi/xxxxx/mount
Failed to create certificate signing request (csr), error is: b'unable to load Private Key\xxxx:error:xxx routines:PEM_read_bio:no start line:pem_lib.c:707:Expecting: ANY PRIVATE KEY\
```

## Reason

`privateKey.pem` that is persisted on the node happens to become an **empty file**, but the [check is not able to detect that](https://github.com/aws/efs-utils/blob/443476f84d03f53f9c11f31a110681ebc31623cf/src/watchdog/__init__.py#L1365) and thus not recreating the key. Hence the node is stale for 1 hour until the cert is is purged.

After efs-csi-node pod restart, note the empty file privateKey.pem when above issues are logged:
```sh
/ # ls -la /host/var/amazon/efs/
-rw-r--r--    1 root     root          2707 Apr 26 07:08 efs-utils.conf
-rw-r--r--    1 root     root          4789 Apr 26 01:16 efs-utils.crt
-rw-r--r--    1 root     root             0 Apr 26 01:17 privateKey.pem
```

Workaround: delete the `privateKey.pem` and restart the pod:
```sh
# ls -la /host/var/amazon/efs/
-rw-r--r--    1 root     root          2707 Apr 26 08:46 efs-utils.conf
-rw-r--r--    1 root     root          4789 Apr 26 01:16 efs-utils.crt
-r--------    1 root     root          2484 Apr 26 08:47 privateKey.pem
```

**What you expected to happen?**
Nodes to stay healthy

**How to reproduce it (as minimally and precisely as possible)?**

same as #178 

**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`): 1.21
- Driver version: 1.3.7

**Comment by universam1** [comment]
@wongma7 provided a potential solution: https://github.com/aws/efs-utils/pull/130

**Comment by NBardelot** [comment]
Somehow we've encountered the bug today with the driver in version 1.3.3 after a rollout where we wanted to restart the driver for all the cluster.

At first it looked like a mount issue (unbound immediate PersistentVolumeClaims).
The socket returning a "connection refused" made us think of a network issue, but our security groups for the EFS Access Points were OK...
It's only after a while that the private key message appears. And it was the issue mentionned here.

Thanks a lot @universam1 for the analysis and the workaround!

**Comment by universam1** [comment]
/remove-lifecycle stale
Kindly asking for review @wongma7

**Comment by stijnbrouwers** [comment]
Would be nice if this could be picked up since we are also facing this...

**Comment by bentatham** [comment]
According to [efs-utils #130](https://github.com/aws/efs-utils/pull/130), this has been fixed in [v1.4.9 of aws-efs-csi-driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver/releases/tag/v1.4.9).

**Comment by headyj** [comment]
same here, seems that sometime the privateKey.pem is empty

**Comment by otorreno** [comment]
I think this PR to efs-utils should fix this issue https://github.com/aws/efs-utils/pull/174/files. We are still facing the issue even with the initContainer.

v1.4.9 didn't fix it completely because the fix was only applied to the watchdog folder, and the same code is duplicated in the mount_efs one

**Comment by Dohbedoh** [comment]
https://github.com/aws/efs-utils/pull/174 is released in efs-utils 1.35.1. And efs-utils 1.35.1 is included in [aws-efs-csi-driver 1.7.4](
https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/e002150e7fc0a6b8448c0813cf4da0295a720af0/CHANGELOG-1.x.md#v174).
